### ISSUE TITLE
linkIdentity() does not work with React Native

### ISSUE THREAD

**Issue by norenz92** [supabase/auth#1645]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Not sure if this is a bug or just not compatible with React Native, but there is no way to use `linkIdentity()` with native social logins like Google, Facebook and Apple in a React Native app. Documentation does not provide any information on how to implement this.

## System information

- OS: macOS
- Version of supabase-js: 2.44.2
- Version of Node.js: v20.5.1

**Comment by kangmingtay** [comment]
> no way to use linkIdentity() with native social logins like Google, Facebook and Apple in a React Native app

@norenz92 can you elaborate more on this statement? do you get some sort of error when you call `linkIdentity` in a react native environment?

**Comment by norenz92** [comment]
> > no way to use linkIdentity() with native social logins like Google, Facebook and Apple in a React Native app
> 
> @norenz92 can you elaborate more on this statement? do you get some sort of error when you call `linkIdentity` in a react native environment?

No error. It's just that this function does not seem to be meant to use with native auth for these providers. The function returns a redirect URL which from AFAIK has no use in my case.

**Comment by kangmingtay** [comment]
@norenz92 it works just like `signInWithOAuth()`, where you need to redirect the user to the URL returned

**Comment by norenz92** [comment]
> @norenz92 it works just like `signInWithOAuth()`, where you need to redirect the user to the URL returned

No, that's not how the native sign in flow works in React Native unfortunately.

**Comment by f-bog** [comment]
@norenz92 I was confused in a similar way, you will need to utilise something like expo-web-browser to prompt open an in-app browser to handle the redirecting. This seems to be the standard way to handle social logins inside of apps.. I'm not aware of a different solution.

**Comment by zwenza** [comment]
Linking works with the standard OAuth flow over the web-browser, however Supabase already has capabilities to support native sign-in methods on mobile platforms.

Currently I can already offer a native sign-in this way using `expo-apple-authentication`:
```
const credential = await AppleAuthentication.signInAsync({
  requestedScopes: [AppleAuthentication.AppleAuthenticationScope.EMAIL],
});

// This works for non-anonymous signups
await supabase.auth.signInWithIdToken({
  provider: "apple",
  token: credential.identityToken,
});

// This is how I would imagine linkIdentity to support native sign-ins 
 const { data, error } = await supabase.auth.linkIdentity({
  provider: "apple",
  token: credential.identityToken,   // This is currently not possible!
  options: { skipBrowserRedirect: true },
 });
```

Is there anything speaking against allowing us to pass the OIDC token to `linkIdentity` like we can already do with `signInWithIdToken`?

**Comment by christophemenager** [comment]
+1 on this! Having the possiblity to `linkIdentity` with a token property would be amazing. Linked issue https://github.com/orgs/supabase/discussions/29017

**Comment by ndrkltsk** [comment]
Any update on this?
